execution/executor-benchmark/src/fake_executor.rs

```diff
@@ -106,110 +122,207 @@ impl FakeExecutor {
+                assert!(
+                    DbAccessUtil::get_coin_store(&recepient_coin_store_key, state_view)?.is_none()
+                );
             }
 
-            let new_account = Account {
-                authentication_key: new_account_address.to_vec(),
+            let recepient_account = Account {
+                authentication_key: recepient_address.to_vec(),
                 ..Default::default()
             };
 
-            let new_coin_store = CoinStore {
-                coin: initial_balance,
+            let recepient_coin_store = CoinStore {
+                coin: transfer_amount,
                 ..Default::default()
             };
 
             write_set.push((
-                new_account_key,
-                WriteOp::Creation(bcs::to_bytes(&new_account)?),
+                recepient_account_key,
+                WriteOp::Creation(bcs::to_bytes(&recepient_account)?),
             ));
             write_set.push((
-                new_coin_store_key,
-                WriteOp::Creation(bcs::to_bytes(&new_coin_store)?),
+                recepient_coin_store_key,
+                WriteOp::Creation(bcs::to_bytes(&recepient_coin_store)?),
             ));
         }
 
-        // TODO(grao): Some values are fake, because I'm lazy.
-        let events = vec![
-            ContractEvent::new(
-                EventKey::new(0, sender_address),
-                0,
-                TypeTag::Struct(Box::new(WithdrawEvent::struct_tag())),
-                sender_address.to_vec(),
-            ),
-            ContractEvent::new(
-                EventKey::new(0, new_account_address),
-                0,
-                TypeTag::Struct(Box::new(DepositEvent::struct_tag())),
-                new_account_address.to_vec(),
-            ),
-            // TODO(grao): CoinRegisterEvent
-        ];
-        Ok(TransactionOutput::new(
-            WriteSetMut::new(write_set).freeze()?,
-            events,
-            /*gas_used=*/ gas,
-            TransactionStatus::Keep(ExecutionStatus::Success),
-        ))
+        let events = vec![ContractEvent::new(
+            EventKey::new(0, recepient_address),
+            0,
+            TypeTag::Struct(Box::new(DepositEvent::struct_tag())),
+            recepient_address.to_vec(),
+        )];
+        Ok(Ok(IncrementalOutput { write_set, events }))
+    }
+
+    fn handle_account_creation_and_transfer(
+        sender_address: AccountAddress,
+        recepient_address: AccountAddress,
+        transfer_amount: u64,
+        state_view: &CachedStateView,
+        fail_on_existing: bool,
+        fail_on_missing: bool,
+    ) -> Result<TransactionOutput> {
+        let _timer = TIMER.with_label_values(&["account_creation"]).start_timer();
+
+        let mut output = {
+            let output = Self::withdraw_from_signer(sender_address, transfer_amount, state_view)?;
+            match output {
+                Ok(output) => output,
```

Comment:
use `Option::or_else()`?

Reply:
unfortunately, I return from the top level function in the "else", so I cannot do that easily

a little bit too nested with Result<Result<IncrementalOutput, TransactionStatus>>, so having two different Err types basically